### ISSUE TITLE
maxdisks synoinfo and cmdline setting is not respected

### ISSUE THREAD

**Issue by cemarriott** [fbelavenuto/arpl#676]
Currently running ARPL 1.1-beta2a with the DS3622xs+ model. I believe this model defaults to maxdisks of 12, however the /etc/synoinfo.conf and /etc.defaults/synoinfo.conf have maxdisks=8. I have 9 drives connected and with maxdisks set this way, the 9th drive does not show up in Storage Manager. 

I have tried adding a synoinfo line via menu.sh, as well as cmdline, but neither are reflected in the final synoinfo.conf file once DSM is started. 

I have tried extracting the initrd-dsm file and checking the synoinfo file there. It does show maxdisks=12, so it seems to reflect correctly in the ramdisk... but something is happening after the ramdisk that must be overwriting it with maxdisks=8? 

Any thoughts on where I can look to troubleshoot this? 
I see an older version of ARPL had a "set maxdisks manually" entry in menu.sh, but it seems that has been removed in more recent versions. 

<img width="1027" alt="Screenshot 2023-03-18 at 5 47 42 PM" src="https://user-images.githubusercontent.com/1731607/226142351-e6f6c313-5884-4163-aab4-1044ec5ada8a.png">

<img width="906" alt="Screenshot 2023-03-18 at 5 55 03 PM" src="https://user-images.githubusercontent.com/1731607/226142354-a98ae56c-ab6a-4f04-a39e-3add4cb27f01.png">

**Comment by AuxXxilium** [comment]
it looks like your dsm has a bug. the later arpl doen't need this. it will be set by boot thru disks addon. and this addon counts your hdd's well. can you provide a screenshot from disksmapping where i can see which ports are used?

The Problem is, that dsm only support 8 disks per sata controller. so i need the diskmap/portmapping to tell you more.

**Comment by cemarriott** [comment]
I appreciate the response. Ultimately I got it working. The situation is a bit unique; the motherboard I am using does not have a SATA controller built in; I am using 2 PCIe SAS cards. Port map and disk count show 0 in ARPL, but DSM itself recognizes the controllers and the disks. 

For some reason DSM was holding on to that maxdisks=8 config. I could clearly see the file generated by ARPL had the appropriate maxdisks=12. 

I set up a clean ARPL image and config, and reinstalled DSM. DSM then had the correct maxdisks=12 and all 9 drives were visible in Storage Manager.
